app/models/uphold_connection.rb

```diff
@@ -51,50 +47,62 @@ def prepare_uphold_state_token
     save!
```

Comment:
while we're in here - I have a quick question about the logic to create the `state`. can you confirm that we will generate a new state for every time we attempt to link (whether we succeed in getting a code back or not)?

it looks to me like there may be possible re-use in the case where the user doesn't complete the linking flow

Reply:
Yes, it certainly seems like we could get into an issue where if this was interrupted then we would be reusing the state token. The [state token is generated only once before they click on it](https://github.com/brave-intl/publishers/blob/store-uphold-in-publishers/app/models/uphold_connection.rb#L44).
We've actually seen the codes not work because of the user not completing which is why there is a `CleanStaleUpholdDataJob` which clears out the codes.

During the first refactor to migrate from storing all these properties directly on the Publisher's model I experimented with regenerating every time but it didn't work due to needing to store the code for more than one request., Thinking about this again I think when the user clicks on "Connect to Uphold" they should be directed to a Publisher's controller which then generated a code, then redirects them to uphold which then redirects back to `uphold_verified`

Reply:
Here's the change that generates a new state token every time the user clicks 'Connect'
https://github.com/brave-intl/publishers/pull/2062/commits/58dda10651cfe2b9f3c1316b843a28f6cf86b633